### PR TITLE
add support for Portuguese (Portugal) (PT)

### PR DESCRIPTION
**Pull request by Olos**
### Changes proposed in this pull request:

* Added support for Portuguese from Portugal

### Status

- [X] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### Additional notes

*  I noticed there was already one pull request regarding this, but I changed some more things from the Brazilian Portuguese, namely 
- 19 ('devEzone') to ('dezAnove') - same logic to 16 and 17
- Also some specifications with using "and" ("e") after some words 1200 is ("mil e duzentos") 1210 is ("mil duzentos e dez") - no "e" before the duzentos
- Using long scale instead of short scale
- Currency - add "de" when "milhões" is right before "euros"
- Removed comma separator (from pt_BR) for big composite numbers
- First PR ever... be gentle

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ dist
 .idea/
 *.egg-info
 /.tox
+venv/
```
Comment (shulcsm) [inline .gitignore]: This should be part of your .git/info/exclude, not in tree.

File: num2words/lang_PT.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    def set_high_numwords(self, high):
+        cap = 3 + 6 * len(high)
+
+        for word, n in zip(high, range(cap, 3, -6)):
+            if self.GIGA_SUFFIX:
+                self.cards[10 ** n] = word + self.GIGA_SUFFIX
+
+            if self.MEGA_SUFFIX:
+                self.cards[10 ** (n - 3)] = word + self.MEGA_SUFFIX
+                if self.cards[10 ** (n - 3)] == 'milião':
```
Comment (shulcsm) [inline num2words/lang_PT.py]: This doesn't make any sense, if you are implementing `set_high_numwords` just set the suffixes you need, no need to use class props

**Comment by shulcsm** [comment]
This seems really bad https://github.com/savoirfairelinux/num2words/pull/198/commits/d1370be7ad2703a3801059a9962a7348b617df62#diff-48c05b2e89d4eb8299b0f502c3707165R224
But is probably out of scope for this pull request.

File: .gitignore (hunk 1 above)
Comment (Olos) [reply .gitignore]: Oh sorry, I'll remove it

File: num2words/lang_PT.py (hunk 2 above)
Comment (Olos) [reply num2words/lang_PT.py]: I just used the set_high_numwords from lang_EU but I needed to correct the "milião". 
Is it better if I correct it in merge? Like:
```python
if ntext == 'milião':
    ntext = 'milhão'
```

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,4 @@ build
 dist
 .idea/
 *.egg-info
-/.tox
```
Comment (erozqba) [inline .gitignore]: Why do you remove the empty line at the end of the file?

Comment (Olos) [reply .gitignore]: Must have been when I deleted the venv/ line.
Sorry. I will add it again.

**Comment by Olos** [comment]
Regarding the differences between the way pt-pt and pt-br write the numbers:
* In pt-pt when you have 10<sup>9</sup> you do not write 'um bilião' but 'mil milhões' (we use the long scale instead of the short scale) ;
* in pt-pt we do not write 19 ('dezEnove'), 17 ('dezEssete'), 16 ('dezEsseis') but ('dezAnove') ('dezAssete') ('dezAsseis');
* in pt-pt we do not say 'bilhão' 'trilhão', etc but 'bilião', 'trilião', etc;
* I did not put a comma after thousand, million, etc. as a separator (I have never used it in my life) but that is a detail;
* in pt-br they don't have the 'e' connection in 200200 'duzentos mil, duzentos' but in pt-pt we say 'duzentos mil e duzentos' (they only have it with 200100)
* I did not have the opportunity to change the ordinal code (yet?) because we also say (in theory because we're talking about big numbers here) them differently, according to my research.

Do you think it can be expanded with these issues?

The other pt-pt PR does not solve, if I analyse it correctly, the issues I stated above.

**Comment by shulcsm** [comment]
@Olos You don't have to necessarily use pt-pt class as inheritance base, just refactor it so you can reuse what is common and get rid of obvious duplication

**Comment by Olos** [comment]
I understand that but in this case:
* setup has (?) to be different
* merge has some different specifications so I don't see how it can be done (not saying it can't be done, just that I don't see it)
* same for to_cardinal: the separator is different; the big numbers are different; the regex is different
* to_currency uses the base class implementation (so already refactored?)

If my previous assumptions are correct I can use pt-br for:
* to_ordinal
* to_ordinal_num
* to_year

As they are all the same verbatim.

Is this it?

Thank you.

**Comment by shulcsm** [comment]
@Olos yes, but you keep redefining huge dictionaries in it. For start move them into module constants and import them in your implementation.

**Comment by Olos** [comment]
Sorry for not updating this yet but some things got in the way.
I'll define the dictionaries out of both PT languages and import them to both of them (with the changes) when I have a little time.
Should I do the same with to_ordinal function or that makes no sense?
Thank you

**Comment by Olos** [comment]
Could I just use PT_pt as base for PT_br? I know pt-br was created first here but it makes more sense (at least to me) to do it like this.
I would change the dictionary delete the ordinal things and keep the merge, to_cardinal (calling the Num2Word_EU.to_cardinal instead of super) and to_currency.
Makes sense?

**Comment by erozqba** [comment]
I'm ok with it @Olos, go ahead unless someone else has an inconvenient with it.

**Comment by Olos** [comment]
@erozqba 
Changed it. 
Waiting on feedback!

File: num2words/lang_PT_BR.py [hunk 4]
```diff
@@ -132,7 +72,7 @@ def merge(self, curr, next):
         return (ctext + ntext, cnum * nnum)
 
     def to_cardinal(self, value):
-        result = super(Num2Word_PT_BR, self).to_cardinal(value)
+        result = lang_PT.Num2Word_EU.to_cardinal(self, value)
```
Comment (erozqba) [inline num2words/lang_PT_BR.py]: Is really better to use the `to_cardinal()` implementation on `Num2Word_EU` instead of that the one on `Num2Word_PT`? They are so different the cardinal numbers between `pt` and `pt_BR`?

Comment (Olos) [reply num2words/lang_PT_BR.py]: Both of them use the `Num2Word_EU` implementation and then change the output based on a regex.
I kept the original regex on `pt_BR` but the regex on `pt` is a little different.
In order for everything to work i thought it was better to continue to use `Num2Word_EU` for both of them.

Comment (erozqba) [reply num2words/lang_PT_BR.py]: @Olos is good for me then! I see you still have the PR marked as **Work In Progress**. If you have finished, change the status, please. Also, I will really appreciate if you can check the missed lines on the coverage by the tests and add some tests for this lines, so we increase the coverage to the maximum, https://coveralls.io/builds/19469219/source?filename=num2words%2Flang_PT.py#L134.

Comment (Olos) [reply num2words/lang_PT_BR.py]: @erozqba Added new tests for those lines.
